substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/monitor/MultiThreadedMonitorSupport.java

```diff
@@ -488,47 +486,45 @@ protected static JavaMonitor newMonitorLock() {
     }
 
     protected static boolean isMonitorLock(JavaMonitor lock) {
-        return lock != null && isMonitorLockSynchronizer(SubstrateUtil.cast(lock, Target_java_util_concurrent_locks_ReentrantLock.class).sync);
+        return lock != null && isMonitorLockSynchronizer(lock.getSync());
     }
 
     protected static boolean isMonitorLockSynchronizer(Object obj) {
-        if (obj != null && obj.getClass() == Target_java_util_concurrent_locks_ReentrantLock_NonfairSync.class) {
-            Target_java_util_concurrent_locks_ReentrantLock_NonfairSync sync = SubstrateUtil.cast(obj, Target_java_util_concurrent_locks_ReentrantLock_NonfairSync.class);
-            return sync.objectMonitorCondition != null; // contains marker or actual condition
+        if (obj != null && obj.getClass() == GraalReentrantLock.GraalNonfairSync.class) {
+            GraalReentrantLock.GraalNonfairSync sync = (GraalReentrantLock.GraalNonfairSync) obj;
+            return sync.getGraalConditionObject() != null; // contains marker or actual condition
         }
         return false;
     }
 
-    public ReentrantLock getMonitorForTesting(Object obj) {
+    public JavaMonitor getMonitorForTesting(Object obj) {
         return getOrCreateMonitor(obj, true);
     }
 
-    protected ConditionObject getOrCreateCondition(JavaMonitor monitorLock, boolean createIfNotExisting) {
+    protected GraalAbstractQueuedSynchronizer.GraalConditionObject getOrCreateCondition(JavaMonitor monitorLock, boolean createIfNotExisting) {
```

Comment:
Feel free to use static imports to make this a bit more readable.

Reply:
done!